### ISSUE TITLE
Failing to build

### ISSUE THREAD

**Issue by ArchangeGabriel** [vector-im/riot-web#3912]
### Steps to reproduce

- Download latest source tarball or clone git master
- Run npm install --cache "<path-to-npm-cache>"
- Run npm run build

Fails to build, log below.
```
riot-web@0.9.9 /build/riot/src/riot-web-0.9.9
├─┬ babel-polyfill@6.23.0 
│ ├── babel-runtime@6.23.0  deduped
│ ├── core-js@2.4.1 
│ └── regenerator-runtime@0.10.5 
├─┬ babel-runtime@6.23.0 
│ ├── core-js@2.4.1  deduped
│ └── regenerator-runtime@0.10.5  deduped
├── browser-request@0.3.3 
├── classnames@2.2.5 
├─┬ draft-js@0.8.1 
│ ├─┬ fbjs@0.8.12 
│ │ ├── core-js@1.2.7 
│ │ ├── isomorphic-fetch@2.2.1  deduped
│ │ ├── loose-envify@1.3.1  deduped
│ │ ├── object-assign@4.1.1  deduped
│ │ ├─┬ promise@7.1.1 
│ │ │ └── asap@2.0.5  deduped
│ │ ├── setimmediate@1.0.5 
│ │ └── ua-parser-js@0.7.12  deduped
│ ├── immutable@3.7.6 
│ └── object-assign@4.1.1 
├─┬ extract-text-webpack-plugin@0.9.1 
│ ├── async@1.5.2 
│ └─┬ loader-utils@0.2.17 
│   ├── big.js@3.1.3 
│   ├── emojis-list@2.1.0 
│   ├── json5@0.5.1 
│   └── object-assign@4.1.1  deduped
├── favico.js@0.3.10 
├── filesize@3.5.9 
├── flux@2.0.3 
├── gemini-scrollbar@1.4.3  (git://github.com/matrix-org/gemini-scrollbar.git#b302279810d05319ac5ff1bd34910bff32325c7b)
├── gfm.css@1.1.1 
├── highlight.js@9.11.0 
├─┬ linkifyjs@2.1.4 
│ ├── jquery@3.2.1 
│ ├── react@15.5.4  deduped
│ └── react-dom@15.5.4  deduped
├─┬ matrix-js-sdk@0.7.7 
│ ├── another-json@0.2.0 
│ ├── browser-request@0.3.3  deduped
│ ├── olm@2.2.1  deduped
│ ├── q@1.5.0  deduped
│ └─┬ request@2.81.0 
│   ├── aws-sign2@0.6.0 
│   ├── aws4@1.6.0 
│   ├── caseless@0.12.0 
│   ├─┬ combined-stream@1.0.5 
│   │ └── delayed-stream@1.0.0 
│   ├── extend@3.0.1 
│   ├── forever-agent@0.6.1 
│   ├─┬ form-data@2.1.4 
│   │ ├── asynckit@0.4.0 
│   │ ├── combined-stream@1.0.5  deduped
│   │ └── mime-types@2.1.15  deduped
│   ├─┬ har-validator@4.2.1 
│   │ ├─┬ ajv@4.11.8 
│   │ │ ├── co@4.6.0 
│   │ │ └─┬ json-stable-stringify@1.0.1 
│   │ │   └── jsonify@0.0.0 
│   │ └── har-schema@1.0.5 
│   ├─┬ hawk@3.1.3 
│   │ ├─┬ boom@2.10.1 
│   │ │ └── hoek@2.16.3  deduped
│   │ ├─┬ cryptiles@2.0.5 
│   │ │ └── boom@2.10.1  deduped
│   │ ├── hoek@2.16.3 
│   │ └─┬ sntp@1.0.9 
│   │   └── hoek@2.16.3  deduped
│   ├─┬ http-signature@1.1.1 
│   │ ├── assert-plus@0.2.0 
│   │ ├─┬ jsprim@1.4.0 
│   │ │ ├── assert-plus@1.0.0 
│   │ │ ├── extsprintf@1.0.2 
│   │ │ ├── json-schema@0.2.3 
│   │ │ └─┬ verror@1.3.6 
│   │ │   └── extsprintf@1.0.2  deduped
│   │ └─┬ sshpk@1.13.0 
│   │   ├── asn1@0.2.3 
│   │   ├── assert-plus@1.0.0 
│   │   ├─┬ bcrypt-pbkdf@1.0.1 
│   │   │ └── tweetnacl@0.14.5  deduped
│   │   ├─┬ dashdash@1.14.1 
│   │   │ └── assert-plus@1.0.0 
│   │   ├─┬ ecc-jsbn@0.1.1 
│   │   │ └── jsbn@0.1.1  deduped
│   │   ├─┬ getpass@0.1.7 
│   │   │ └── assert-plus@1.0.0 
│   │   ├─┬ jodid25519@1.0.2 
│   │   │ └── jsbn@0.1.1  deduped
│   │   ├── jsbn@0.1.1 
│   │   └── tweetnacl@0.14.5 
│   ├── is-typedarray@1.0.0 
│   ├── isstream@0.1.2 
│   ├── json-stringify-safe@5.0.1 
│   ├─┬ mime-types@2.1.15 
│   │ └── mime-db@1.27.0 
│   ├── oauth-sign@0.8.2 
│   ├── performance-now@0.2.0 
│   ├── qs@6.4.0 
│   ├── safe-buffer@5.0.1 
│   ├── stringstream@0.0.5 
│   ├─┬ tough-cookie@2.3.2 
│   │ └── punycode@1.4.1 
│   ├─┬ tunnel-agent@0.6.0 
│   │ └── safe-buffer@5.0.1  deduped
│   └── uuid@3.0.1 
├─┬ matrix-react-sdk@0.8.8 
│ ├── babel-runtime@6.23.0  deduped
│ ├── blueimp-canvas-to-blob@3.7.0 
│ ├── browser-encrypt-attachment@0.3.0 
│ ├── browser-request@0.3.3  deduped
│ ├── classnames@2.2.5  deduped
│ ├─┬ commonmark@0.27.0 
│ │ ├── entities@1.1.1 
│ │ ├── mdurl@1.0.1 
│ │ ├── minimist@1.2.0  deduped
│ │ └── string.prototype.repeat@0.2.0 
│ ├── draft-js@0.8.1  deduped
│ ├─┬ draft-js-export-html@0.5.4 
│ │ └── draft-js-utils@0.1.7 
│ ├─┬ draft-js-export-markdown@0.2.2 
│ │ └── draft-js-utils@0.1.7  deduped
│ ├── emojione@2.2.3 
│ ├── file-saver@1.3.3 
│ ├── filesize@3.5.9  deduped
│ ├── flux@2.0.3  deduped
│ ├── fuse.js@2.7.4 
│ ├─┬ glob@5.0.15 
│ │ ├─┬ inflight@1.0.6 
│ │ │ ├── once@1.4.0  deduped
│ │ │ └── wrappy@1.0.2 
│ │ ├── inherits@2.0.3 
│ │ ├─┬ minimatch@3.0.4 
│ │ │ └─┬ brace-expansion@1.1.7 
│ │ │   ├── balanced-match@0.4.2 
│ │ │   └── concat-map@0.0.1 
│ │ ├─┬ once@1.4.0 
│ │ │ └── wrappy@1.0.2  deduped
│ │ └── path-is-absolute@1.0.1 
│ ├── highlight.js@8.9.1 
│ ├─┬ isomorphic-fetch@2.2.1 
│ │ ├─┬ node-fetch@1.6.3 
│ │ │ ├─┬ encoding@0.1.12 
│ │ │ │ └── iconv-lite@0.4.17 
│ │ │ └── is-stream@1.1.0 
│ │ └── whatwg-fetch@2.0.3 
│ ├── linkifyjs@2.1.4  deduped
│ ├── lodash@4.17.4 
│ ├── matrix-js-sdk@0.7.7  deduped
│ ├─┬ optimist@0.6.1 
│ │ ├── minimist@0.0.10 
│ │ └── wordwrap@0.0.3 
│ ├── q@1.5.0  deduped
│ ├── react@15.5.4  deduped
│ ├── react-addons-css-transition-group@15.3.2 
│ ├── react-dom@15.5.4  deduped
│ ├── react-gemini-scrollbar@2.1.5  deduped (git://github.com/matrix-org/react-gemini-scrollbar.git#5e97aef7e034efc8db1431f4b0efe3b26e249ae9)
│ ├── sanitize-html@1.14.1  deduped
│ ├── text-encoding-utf-8@1.0.1 
│ ├─┬ velocity-vector@1.2.3  (git://github.com/vector-im/velocity.git#059e3b2348f1110888d033974d3109fd5a3af00f)
│ │ └── jquery@3.2.1  deduped
│ └── whatwg-fetch@1.1.1 
├── minimist@1.2.0 
├─┬ mkdirp@0.5.1 
│ └── minimist@0.0.8 
├─┬ modernizr@3.5.0 
│ ├─┬ doctrine@1.2.3 
│ │ ├── esutils@2.0.2 
│ │ └── isarray@1.0.0 
│ ├── file@0.2.2 
│ ├── find-parent-dir@0.3.0 
│ ├── lodash@4.17.4  deduped
│ ├── mkdirp@0.5.1  deduped
│ ├─┬ remarkable@1.7.1 
│ │ ├─┬ argparse@0.1.16 
│ │ │ ├── underscore@1.7.0 
│ │ │ └── underscore.string@2.4.0 
│ │ └── autolinker@0.15.3 
│ ├── requirejs@2.1.22 
│ └─┬ yargs@7.0.2 
│   ├── camelcase@3.0.0 
│   ├─┬ cliui@3.2.0 
│   │ ├── string-width@1.0.2  deduped
│   │ ├─┬ strip-ansi@3.0.1 
│   │ │ └── ansi-regex@2.1.1 
│   │ └─┬ wrap-ansi@2.1.0 
│   │   ├── string-width@1.0.2  deduped
│   │   └── strip-ansi@3.0.1  deduped
│   ├── decamelize@1.2.0 
│   ├── get-caller-file@1.0.2 
│   ├─┬ os-locale@1.4.0 
│   │ └─┬ lcid@1.0.0 
│   │   └── invert-kv@1.0.0 
│   ├─┬ read-pkg-up@1.0.1 
│   │ ├─┬ find-up@1.1.2 
│   │ │ ├─┬ path-exists@2.1.0 
│   │ │ │ └── pinkie-promise@2.0.1  deduped
│   │ │ └─┬ pinkie-promise@2.0.1 
│   │ │   └── pinkie@2.0.4 
│   │ └─┬ read-pkg@1.1.0 
│   │   ├─┬ load-json-file@1.1.0 
│   │   │ ├── graceful-fs@4.1.11 
│   │   │ ├─┬ parse-json@2.2.0 
│   │   │ │ └─┬ error-ex@1.3.1 
│   │   │ │   └── is-arrayish@0.2.1 
│   │   │ ├── pify@2.3.0 
│   │   │ ├── pinkie-promise@2.0.1  deduped
│   │   │ └─┬ strip-bom@2.0.0 
│   │   │   └── is-utf8@0.2.1 
│   │   ├─┬ normalize-package-data@2.3.8 
│   │   │ ├── hosted-git-info@2.4.2 
│   │   │ ├─┬ is-builtin-module@1.0.0 
│   │   │ │ └── builtin-modules@1.1.1 
│   │   │ ├── semver@5.3.0 
│   │   │ └─┬ validate-npm-package-license@3.0.1 
│   │   │   ├─┬ spdx-correct@1.0.2 
│   │   │   │ └── spdx-license-ids@1.2.2 
│   │   │   └── spdx-expression-parse@1.0.4 
│   │   └─┬ path-type@1.1.0 
│   │     ├── graceful-fs@4.1.11  deduped
│   │     ├── pify@2.3.0  deduped
│   │     └── pinkie-promise@2.0.1  deduped
│   ├── require-directory@2.1.1 
│   ├── require-main-filename@1.0.1 
│   ├── set-blocking@2.0.0 
│   ├─┬ string-width@1.0.2 
│   │ ├── code-point-at@1.1.0 
│   │ ├─┬ is-fullwidth-code-point@1.0.0 
│   │ │ └── number-is-nan@1.0.1 
│   │ └── strip-ansi@3.0.1  deduped
│   ├── which-module@1.0.0 
│   ├── y18n@3.2.1 
│   └─┬ yargs-parser@5.0.0 
│     └── camelcase@3.0.0  deduped
├── olm@2.2.1 
├── q@1.5.0 
├─┬ react@15.5.4 
│ ├── fbjs@0.8.12  deduped
│ ├─┬ loose-envify@1.3.1 
│ │ └── js-tokens@3.0.1 
│ ├── object-assign@4.1.1  deduped
│ └─┬ prop-types@15.5.10 
│   ├── fbjs@0.8.12  deduped
│   └── loose-envify@1.3.1  deduped
├─┬ react-dnd@2.4.0 
│ ├── disposables@1.0.1 
│ ├─┬ dnd-core@2.4.0 
│ │ ├── asap@2.0.5 
│ │ ├── invariant@2.2.2  deduped
│ │ ├── lodash@4.17.4  deduped
│ │ └─┬ redux@3.6.0 
│ │   ├── lodash@4.17.4  deduped
│ │   ├── lodash-es@4.17.4 
│ │   ├── loose-envify@1.3.1  deduped
│ │   └── symbol-observable@1.0.4 
│ ├── hoist-non-react-statics@1.2.0 
│ ├─┬ invariant@2.2.2 
│ │ └── loose-envify@1.3.1  deduped
│ ├── lodash@4.17.4  deduped
│ └── prop-types@15.5.10  deduped
├─┬ react-dnd-html5-backend@2.4.1 
│ └── lodash@4.17.4  deduped
├─┬ react-dom@15.5.4 
│ ├── fbjs@0.8.12  deduped
│ ├── loose-envify@1.3.1  deduped
│ ├── object-assign@4.1.1  deduped
│ └── prop-types@15.5.10  deduped
├─┬ react-gemini-scrollbar@2.1.5  (git://github.com/matrix-org/react-gemini-scrollbar.git#5e97aef7e034efc8db1431f4b0efe3b26e249ae9)
│ └── gemini-scrollbar@1.4.3  deduped (git://github.com/matrix-org/gemini-scrollbar.git#b302279810d05319ac5ff1bd34910bff32325c7b)
├─┬ sanitize-html@1.14.1 
│ ├─┬ htmlparser2@3.9.2 
│ │ ├── domelementtype@1.3.0 
│ │ ├─┬ domhandler@2.4.1 
│ │ │ └── domelementtype@1.3.0  deduped
│ │ ├─┬ domutils@1.6.2 
│ │ │ ├─┬ dom-serializer@0.1.0 
│ │ │ │ ├── domelementtype@1.1.3 
│ │ │ │ └── entities@1.1.1  deduped
│ │ │ └── domelementtype@1.3.0  deduped
│ │ ├── entities@1.1.1  deduped
│ │ ├── inherits@2.0.3  deduped
│ │ └─┬ readable-stream@2.2.9 
│ │   ├── buffer-shims@1.0.0 
│ │   ├── core-util-is@1.0.2 
│ │   ├── inherits@2.0.3  deduped
│ │   ├── isarray@1.0.0  deduped
│ │   ├── process-nextick-args@1.0.7 
│ │   ├─┬ string_decoder@1.0.0 
│ │   │ └── buffer-shims@1.0.0  deduped
│ │   └── util-deprecate@1.0.2 
│ ├── regexp-quote@0.0.0 
│ └── xtend@4.0.1 
├── ua-parser-js@0.7.12 
└─┬ url@0.11.0 
  ├── punycode@1.3.2 
  └── querystring@0.2.0 

npm WARN extract-text-webpack-plugin@0.9.1 requires a peer of webpack@^1.9.11 but none was installed.

> riot-web@0.9.9 build /build/riot/src/riot-web-0.9.9
> node scripts/babelcheck.js && npm run build:res && npm run build:bundle


> riot-web@0.9.9 build:res /build/riot/src/riot-web-0.9.9
> node scripts/copy-res.js

module.js:472
    throw err;
    ^

Error: Cannot find module 'cpx'
    at Function.Module._resolveFilename (module.js:470:15)
    at Function.Module._load (module.js:418:25)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/build/riot/src/riot-web-0.9.9/scripts/copy-res.js:17:13)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! riot-web@0.9.9 build:res: `node scripts/copy-res.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the riot-web@0.9.9 build:res script 'node scripts/copy-res.js'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the riot-web package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node scripts/copy-res.js
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs riot-web
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls riot-web
npm ERR! There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /build/riot/src/npm-cache/_logs/2017-05-14T22_11_04_466Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! riot-web@0.9.9 build: `node scripts/babelcheck.js && npm run build:res && npm run build:bundle`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the riot-web@0.9.9 build script 'node scripts/babelcheck.js && npm run build:res && npm run build:bundle'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the riot-web package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node scripts/babelcheck.js && npm run build:res && npm run build:bundle
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs riot-web
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls riot-web
npm ERR! There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /build/riot/src/npm-cache/_logs/2017-05-14T22_11_04_474Z-debug.log
```

First mentioned log:
```
0 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/node', '/usr/bin/npm', 'run', 'build:res' ]
2 info using npm@4.5.0
3 info using node@v7.10.0
4 verbose run-script [ 'prebuild:res', 'build:res', 'postbuild:res' ]
5 info lifecycle riot-web@0.9.9~prebuild:res: riot-web@0.9.9
6 silly lifecycle riot-web@0.9.9~prebuild:res: no script for prebuild:res, continuing
7 info lifecycle riot-web@0.9.9~build:res: riot-web@0.9.9
8 verbose lifecycle riot-web@0.9.9~build:res: unsafe-perm in lifecycle true
9 verbose lifecycle riot-web@0.9.9~build:res: PATH: /usr/lib/node_modules/npm/bin/node-gyp-bin:/build/riot/src/riot-web-0.9.9/node_modules/.bin:/usr/lib/node_modules/npm/bin/node-gyp-bin:/build/riot/src/riot-web-0.9.9/node_modules/.bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl
10 verbose lifecycle riot-web@0.9.9~build:res: CWD: /build/riot/src/riot-web-0.9.9
11 silly lifecycle riot-web@0.9.9~build:res: Args: [ '-c', 'node scripts/copy-res.js' ]
12 silly lifecycle riot-web@0.9.9~build:res: Returned: code: 1  signal: null
13 info lifecycle riot-web@0.9.9~build:res: Failed to exec build:res script
14 verbose stack Error: riot-web@0.9.9 build:res: `node scripts/copy-res.js`
14 verbose stack Exit status 1
14 verbose stack     at EventEmitter.<anonymous> (/usr/lib/node_modules/npm/lib/utils/lifecycle.js:279:16)
14 verbose stack     at emitTwo (events.js:106:13)
14 verbose stack     at EventEmitter.emit (events.js:194:7)
14 verbose stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/npm/lib/utils/spawn.js:40:14)
14 verbose stack     at emitTwo (events.js:106:13)
14 verbose stack     at ChildProcess.emit (events.js:194:7)
14 verbose stack     at maybeClose (internal/child_process.js:899:16)
14 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)
15 verbose pkgid riot-web@0.9.9
16 verbose cwd /build/riot/src/riot-web-0.9.9
17 verbose Linux 4.10.13-1-ARCH
18 verbose argv "/usr/bin/node" "/usr/bin/npm" "run" "build:res"
19 verbose node v7.10.0
20 verbose npm  v4.5.0
21 error code ELIFECYCLE
22 error errno 1
23 error riot-web@0.9.9 build:res: `node scripts/copy-res.js`
23 error Exit status 1
24 error Failed at the riot-web@0.9.9 build:res script 'node scripts/copy-res.js'.
24 error Make sure you have the latest version of node.js and npm installed.
24 error If you do, this is most likely a problem with the riot-web package,
24 error not with npm itself.
24 error Tell the author that this fails on your system:
24 error     node scripts/copy-res.js
24 error You can get information on how to open an issue for this project with:
24 error     npm bugs riot-web
24 error Or if that isn't available, you can get their info via:
24 error     npm owner ls riot-web
24 error There is likely additional logging output above.
25 verbose exit [ 1, true ]
```

Second one:
```
0 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/node',
1 verbose cli   '/usr/bin/npm',
1 verbose cli   'run',
1 verbose cli   'build',
1 verbose cli   '--cache',
1 verbose cli   '/build/riot/src/npm-cache' ]
2 info using npm@4.5.0
3 info using node@v7.10.0
4 verbose run-script [ 'prebuild', 'build', 'postbuild' ]
5 info lifecycle riot-web@0.9.9~prebuild: riot-web@0.9.9
6 silly lifecycle riot-web@0.9.9~prebuild: no script for prebuild, continuing
7 info lifecycle riot-web@0.9.9~build: riot-web@0.9.9
8 verbose lifecycle riot-web@0.9.9~build: unsafe-perm in lifecycle true
9 verbose lifecycle riot-web@0.9.9~build: PATH: /usr/lib/node_modules/npm/bin/node-gyp-bin:/build/riot/src/riot-web-0.9.9/node_modules/.bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl
10 verbose lifecycle riot-web@0.9.9~build: CWD: /build/riot/src/riot-web-0.9.9
11 silly lifecycle riot-web@0.9.9~build: Args: [ '-c',
11 silly lifecycle   'node scripts/babelcheck.js && npm run build:res && npm run build:bundle' ]
12 silly lifecycle riot-web@0.9.9~build: Returned: code: 1  signal: null
13 info lifecycle riot-web@0.9.9~build: Failed to exec build script
14 verbose stack Error: riot-web@0.9.9 build: `node scripts/babelcheck.js && npm run build:res && npm run build:bundle`
14 verbose stack Exit status 1
14 verbose stack     at EventEmitter.<anonymous> (/usr/lib/node_modules/npm/lib/utils/lifecycle.js:279:16)
14 verbose stack     at emitTwo (events.js:106:13)
14 verbose stack     at EventEmitter.emit (events.js:194:7)
14 verbose stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/npm/lib/utils/spawn.js:40:14)
14 verbose stack     at emitTwo (events.js:106:13)
14 verbose stack     at ChildProcess.emit (events.js:194:7)
14 verbose stack     at maybeClose (internal/child_process.js:899:16)
14 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)
15 verbose pkgid riot-web@0.9.9
16 verbose cwd /build/riot/src/riot-web-0.9.9
17 verbose Linux 4.10.13-1-ARCH
18 verbose argv "/usr/bin/node" "/usr/bin/npm" "run" "build" "--cache" "/build/riot/src/npm-cache"
19 verbose node v7.10.0
20 verbose npm  v4.5.0
21 error code ELIFECYCLE
22 error errno 1
23 error riot-web@0.9.9 build: `node scripts/babelcheck.js && npm run build:res && npm run build:bundle`
23 error Exit status 1
24 error Failed at the riot-web@0.9.9 build script 'node scripts/babelcheck.js && npm run build:res && npm run build:bundle'.
24 error Make sure you have the latest version of node.js and npm installed.
24 error If you do, this is most likely a problem with the riot-web package,
24 error not with npm itself.
24 error Tell the author that this fails on your system:
24 error     node scripts/babelcheck.js && npm run build:res && npm run build:bundle
24 error You can get information on how to open an issue for this project with:
24 error     npm bugs riot-web
24 error Or if that isn't available, you can get their info via:
24 error     npm owner ls riot-web
24 error There is likely additional logging output above.
25 verbose exit [ 1, true ]
```

Please ask if you need any supplemental information, if I should try specific steps, etc.

**Comment by t3chguy** [comment]
For whatever reason your `npm i` isn't installing `cpx` which it definitely should:
https://github.com/vector-im/riot-web/blob/master/package.json#L95

**Comment by ArchangeGabriel** [comment]
Hum, more generally it seems to be skipping the `devDependencies` category. So I’ve removed the --production arg after reading that this was its effect (I think I add already tried that hence why I did not included it above, but apparently I didn’t save the file), and it worked. I’m just unsure whether system phantomjs was used or not, but that’s not a big matter.

Thanks for your help!

**Comment by rkfg** [comment]
This is the only bug that I could find by "copy-res.js" so my advice should be related. In my case it was:
```
npm ERR! Tell the author that this fails on your system:
npm ERR!     node scripts/copy-res.js -w
```
I ran that command, `node scripts/copy-res.js -w`, and it said: `watch /home/rkfg/soft/svn-soft/riot-web/res/flags ENOSPC` which is very misleading because I have plenty of space on each drive and /tmp. It turned out it ran out of inotify watches. I have a lot of files in `node_modules`, about 30k, and max watches is only 8192. So I ran `sysctl fs.inotify.max_user_watches=65536` (add the setting to /etc/sysctl.conf or sysctl.d/some_local.conf for permanent effect) and it worked fine after that. Hope this will help someone.
